src/deepsparse/loggers/config.py

```diff
@@ -0,0 +1,151 @@
+]
+
+
+def get_function_and_function_name(
+    function_identifier: str,
+) -> Tuple[Callable[[Any], Any], str]:
+    """
+    Parse function identifier and return the function as well as its name
+
+    :param function_identifier: Can be one of the following:
+
+        1. framework function, e.g.
+            "torch.mean" or "numpy.max"
+
+        2. built-in function, e.g.
+            "builtins:function_name"
+
+        3. user-defined function in the form of
+           '<path_to_the_python_script>:<function_name>', e.g.
+           "{...}/script_name.py:function_name"
+
+    :return: A tuple (function, function name)
+    """
+
+    if function_identifier.startswith("torch."):
+        func_name = function_identifier.split(".")[1]
+        return getattr(torch, func_name), func_name
+
+    if function_identifier.startswith("numpy.") or function_identifier.startswith(
+        "np."
+    ):
+        func_name = function_identifier.split(".")[1]
+        return getattr(numpy, func_name), func_name
+
+    if function_identifier.startswith("builtins:"):
+        func_name = function_identifier.split(":")[1]
+        return getattr(built_ins, func_name), func_name
+
+    # assume a dynamic import function of the form
+    # '<path_to_the_python_script>:<function_name>'
+    path, func_name = function_identifier.split(":")
+    spec = importlib.util.spec_from_file_location("user_defined_metric_functions", path)
+    module = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(module)
+    return getattr(module, func_name), func_name
+
+
+class MetricFunctionConfig(BaseModel):
+    """
+    Holds logging configuration for a metric function
+    """
+
+    func: str = Field(description="Identifier of the metric function")
+    function: Callable[[Any], Any] = Field(description="The metric function callable")
+    function_name: str = Field(description="Name of the metric function")
+    frequency: int = Field(
+        default=1,
+        description="Specifies how often the function should be applied"
+        "(measured in numbers of inference calls",
+    )
+    logger: Optional[List[str]] = Field(
+        default=None,
+        description="Overrides the global logger configuration in "
+        "the context of the DeepSparse server. "
+        "If not None, this configuration stops logging data "
+        "to globally specified loggers, and will only use "
+        "the subset of loggers "
+        "(specified by a list of their names)",
+    )
+
+    def __init__(self, **data):
+        if data.get("function"):
+            raise ValueError()
+        if data.get("function_name"):
+            raise ValueError()
+        function_identifier = data.get("func")
+        # automatically extract function and function name
+        # from the function_identifier
+        function, function_name = get_function_and_function_name(function_identifier)
+        data["function"], data["function_name"] = function, function_name
```

Comment:
This is parsing code - technically different from configuration, could go into a different place in the code. Just noting here while reviewing

Reply:
Good call, I like that